collectors/proc.plugin/sys_fs_btrfs.c

```diff
@@ -736,6 +1020,178 @@ int do_sys_fs_btrfs(int update_every, usec_t dt) {
+            }
+
+            rrddim_set_by_pointer(node->st_commit_timings, node->rd_commit_timings_last, node->commit_timings_last);
+            rrddim_set_by_pointer(node->st_commit_timings, node->rd_commit_timings_max, node->commit_timings_max);
+            rrdset_done(node->st_commit_timings);
+        }
+
+        // --------------------------------------------------------------------
+        // error_stats per device
+
+        if(do_error_stats == CONFIG_BOOLEAN_YES ||  (do_error_stats == CONFIG_BOOLEAN_AUTO &&
+                                                    (node->devices ||
+                                                    netdata_zero_metrics_enabled == CONFIG_BOOLEAN_YES))) {
+            do_error_stats = CONFIG_BOOLEAN_YES;
+
+            if(unlikely(!node->st_error_stats)) {
+                char id[RRD_ID_LENGTH_MAX + 1], name[RRD_ID_LENGTH_MAX + 1], title[200 + 1];
+
+                snprintf(id, RRD_ID_LENGTH_MAX, "error_stats_%s", node->id);
+                snprintf(name, RRD_ID_LENGTH_MAX, "error_stats_%s", node->label);
+                snprintf(title, 200, "BTRFS Errors");
+
+                netdata_fix_chart_id(id);
+                netdata_fix_chart_name(name);
+
+                node->st_error_stats = rrdset_create_localhost(
+                        "btrfs"
+                        , id
+                        , name
+                        , node->label
+                        , "btrfs.errors"
+                        , title
+                        , "errors"
+                        , PLUGIN_PROC_NAME
+                        , PLUGIN_PROC_MODULE_BTRFS_NAME
+                        , NETDATA_CHART_PRIO_BTRFS_ERRORS
+                        , update_every
+                        , RRDSET_TYPE_LINE
+                );
+
+                add_labels_to_btrfs(node, node->st_error_stats);
+            }
+
+
+            for(BTRFS_DEVICE *d = node->devices ; d ; d = d->next) {
+                char rd_id[RRD_ID_LENGTH_MAX + 1];
+                if(unlikely(!d->rd_write_errs)){
+                    snprintfz(rd_id, RRD_ID_LENGTH_MAX, "%d:write_errs", d->id);
+                    d->rd_write_errs = rrddim_add(node->st_error_stats, rd_id, NULL, 1, 1, RRD_ALGORITHM_ABSOLUTE);
+                    snprintfz(rd_id, RRD_ID_LENGTH_MAX, "%d:read_errs", d->id);
+                    d->rd_read_errs = rrddim_add(node->st_error_stats, rd_id, NULL, 1, 1, RRD_ALGORITHM_ABSOLUTE);
+                    snprintfz(rd_id, RRD_ID_LENGTH_MAX, "%d:flush_errs", d->id);
+                    d->rd_flush_errs = rrddim_add(node->st_error_stats, rd_id, NULL, 1, 1, RRD_ALGORITHM_ABSOLUTE);
+                    snprintfz(rd_id, RRD_ID_LENGTH_MAX, "%d:corruption_errs", d->id);
+                    d->rd_corruption_errs = rrddim_add(node->st_error_stats, rd_id, NULL, 1, 1, RRD_ALGORITHM_ABSOLUTE);
+                    snprintfz(rd_id, RRD_ID_LENGTH_MAX, "%d:generation_errs", d->id);
```

Comment:
I think you need to create a chart per device, not a set of dimensions per device. If you do it like this these charts will be perfectly aggregated.

Reply:
Hmm, good point about aggregation, one chart per device is probably the best option here.

Reply:
Yeah, if we keep it as it is - the charts are not aggregatable.

Reply:
I can only repeat that dimension "per device per X (read/write errors in your case)" is wrong, should be a chart per device (if I understand correctly, and `1` and `2` are devices).

Reply:
I stand corrected, I had misunderstood how grouping works on the cloud (and that error charts can be grouped by context in this particular case). In which case what you said makes total sense. 

I pushed a new commit addressing this.